### ISSUE TITLE
How to force a rebuild of the namespaces/repositories list?

### ISSUE THREAD

**Issue by marc0s** [SUSE/Portus#1870]
I have `portus:head` running with docker compose and after pulling a new version of the image, running `rake db:migrate` and restarting, I have no namespaces nor repositories available in the UI.  I expected to have them back available by forcing the background service to run with `strategy: update-delete`, but nothing appeared.

Is there some command to be run to force such syncing that I may have missed?  Is this a bug instead?

Thanks!

**Comment by marc0s** [comment]
I think that the problem is not the missing data.

Further checks revealed that data is in the DB (tags, namespaces, ...) and that the UI behaved errant: couldn't log out, couldn't check the list of issued application tokens, ...

The browser's console spits out an error on missing jQuery.

```
rails.js:555 Uncaught ReferenceError: jQuery is not defined
    at Object.<anonymous> (rails.js:555)
    at t (bootstrap 63e01a5…:19)
    at Object.<anonymous> (application-9581569e…a254.js? [sm]:39679)
    at t (bootstrap 63e01a5…:19)
    at bootstrap 63e01a5…:65
    at bootstrap 63e01a5…:65
```

Could it be related to https://github.com/SUSE/Portus/pull/1811/files ?  Any workaround you can think of in the meantime?

Thanks!

**Comment by vitoravelino** [comment]
The docker images [0] are not being built for a while and it's possible that we got a broken image for `head`. I can ensure you that `master` is working as expected [1].

You could try `2.3` tag or build the image on your own based on the `master` branch until the docker images build is fixed. 

[0] https://hub.docker.com/r/opensuse/portus/builds/
[1] 
![screenshot-20180709103022-985x258](https://user-images.githubusercontent.com/188554/42453440-20a0f47e-8363-11e8-9f01-702906f5d64b.png)

**Comment by marc0s** [comment]
Thanks for the feedback @vitoravelino.

Will try to build it by myself.  Wouldn't I succeed, is downgrade supported or at least expected to work?

**Comment by vitoravelino** [comment]
Well, it's supposed to work. If something is not working, we should fix it and release a new patch version. Let us know.

**Comment by marc0s** [comment]
I have built a Docker image from this repository's `./docker/Dockerfile` by cloning this repo and running `docker build .` (from `./docker`).  And I'm hitting the exact same error from above.

I updated my `docker-compose.yml` file to use the image generated by that `Dockerfile` and run:

```
$ docker-compose stop portus background
$ docker-compose up -d --force-recreate portus background
```

Any extra step should I run?

**Comment by marc0s** [comment]
By downgrading to `2.3` I almost have it working.  Everything seems to work OK except listing tags inside repository browsing. Portus logs show:

```
portus_1      | 2018-07-09T15:31:31.480281956Z caught error of type NoMethodError in after callback inside Grape::Middleware::Formatter : API::Entities::Tags missing attribute `vulnerabilities' on #<Tag:0x0000000005a44fe8>
portus_1      | 2018-07-09T15:31:31.480299794Z caught error of type NoMethodError in after callback inside Grape::Middleware::Formatter : API::Entities::Tags missing attribute `vulnerabilities' on #<Tag:0x0000000005a762c8>
```

Any migration step to run apart from `rake db:migrate`?

**Comment by vitoravelino** [comment]
What I've done in the past was copying docker[-compose] files to the root folder and running `docker build -t opensuse/portus:head .`.

**Comment by vitoravelino** [comment]
About the vulnerabilities error, I recommend you to restart everything. That error should not happen because this attribute is available for both `2.3` and `head`. 

Also, be aware that if you've downgraded to `2.3` with an existent db from `head` that might break things.

**Comment by marc0s** [comment]
I think I'd need some guidance on getting the image built.  The image I build from `docker/Dockerfile` still shows the error about the missing jquery.

I'm not a SUSE guy, so I barely understand that the `portus` package gets installed in the docker image (I checked that the downloaded one is `2.4.0-dev...`) but I don't know the extra steps to get assets ready (where I suppose jQuery should be available from).

Tried `docker-compose exec portus portusctl exec 'rake portus:assets:compile'` but it complains about missing `font-awesome`... I'm probably doing something plain wrong.

Any help would be appreciated :)
